src/lib/routes/admin-api/api-token-controller.ts

```diff
@@ -0,0 +1,141 @@
+import { Response } from 'express';
+
+import Controller from '../controller';
+import { ADMIN, CREATE_API_TOKEN, DELETE_API_TOKEN, UPDATE_API_TOKEN } from '../../permissions';
+import { ApiTokenService } from '../../services/api-token-service';
+import { Logger, LogProvider } from '../../logger';
+import { ApiTokenType } from '../../db/api-token-store';
+import { AccessService } from '../../services/access-service';
+import { IAuthRequest } from '../unleash-types';
+import { isRbacEnabled } from '../../util/feature-enabled';
+import User from '../../user';
+
+interface IExperimentalFlags {
+    [key: string]: boolean;
+}
+
+interface IConfig {
+    getLogger: LogProvider;
+    extendedPermissions: boolean;
+    experimental: IExperimentalFlags;
+}
+
+interface IServices {
+    apiTokenService: ApiTokenService;
+    accessService: AccessService;
+}
+
+class ApiTokenController extends Controller {
+    private apiTokenService: ApiTokenService;
+
+    private accessService: AccessService;
+
+    private extendedPermissions: boolean;
+
+    private isRbacEnabled: boolean;
+
+    private logger: Logger;
+
+    constructor(config: IConfig, services: IServices) {
+        super(config);
+        this.apiTokenService = services.apiTokenService;
+        this.accessService = services.accessService;
+        this.extendedPermissions = config.extendedPermissions;
+        this.isRbacEnabled = isRbacEnabled(config);
+        this.logger = config.getLogger('api-token-controller.js');
+
+        this.get('/', this.getAllApiTokens);
+        this.post('/', this.createApiToken, CREATE_API_TOKEN);
+        this.put('/:token', this.updateApiToken, UPDATE_API_TOKEN);
+        this.delete('/:token', this.deleteApiToken, DELETE_API_TOKEN);
+    }
+
+    private isTokenAdmin(user: User) {
```

Comment:
The purpose of this method is to allow us to check if a user is allowed to manage API tokens. If yes they should also be allowed to see admin tokens.